craft_parts/packages/snaps.py

```diff
@@ -0,0 +1,393 @@
+            ) from err
+
+        # Now that the snap is refreshed, invalidate the data we had on it.
+        self._is_installed = None
+
+
+def download_snaps(*, snaps_list: Sequence[str], directory: str) -> None:
+    """Download snaps of the format <snap-name>/<channel> into directory.
+
+    The target directory is created if it does not exist.
+    """
+    # TODO manifest.yaml with snap revision from future machine output
+    # for `snap download`.
+    os.makedirs(directory, exist_ok=True)
+    for snap in snaps_list:
+        snap_pkg = SnapPackage(snap)
+
+        # TODO: use dependency injected echoer
+        logger.info("Downloading snap %s", snap_pkg.name)
+        snap_pkg.download(directory=directory)
+
+
+def install_snaps(snaps_list: Union[Sequence[str], Set[str]]) -> List[str]:
+    """Install snaps of the format <snap-name>/<channel>.
+
+    :return: a list of "name=revision" for the snaps installed.
+    """
+    snaps_installed = []
+    for snap in snaps_list:
+        snap_pkg = SnapPackage(snap)
+
+        # Allow bases to be installed from non stable channels.
+        snap_pkg_channel = snap_pkg.get_store_snap_info()["channel"]
+        snap_pkg_type = snap_pkg.get_store_snap_info()["type"]
+        if snap_pkg_channel != "stable" and snap_pkg_type == "base":
+            snap_pkg = SnapPackage(
+                "{snap_name}/latest/{channel}".format(
+                    snap_name=snap_pkg.name, channel=snap_pkg_channel
+                )
+            )
+
+        if not snap_pkg.installed:
+            snap_pkg.install()
+        elif snap_pkg.get_current_channel() != snap_pkg.channel:
+            snap_pkg.refresh()
+
+        snaps_installed.append(
+            "{}={}".format(snap_pkg.name, snap_pkg.get_local_snap_info()["revision"])
+        )
+    return snaps_installed
+
+
+def _snap_command_requires_sudo():
+    # snap whoami returns - if the user is not logged in.
+    output = check_output(["snap", "whoami"])
+    whoami = output.decode(sys.getfilesystemencoding())
+    requires_root = False
+    try:
+        requires_root = whoami.split(":")[1].strip() == "-"
+    # A safeguard if the output changes
+    except IndexError:
+        requires_root = True
+    if requires_root:
+        logger.warning("snapd is not logged in, snap install commands will use sudo")
+    return requires_root
+
+
+def get_assertion(assertion_params: Sequence[str]) -> bytes:
+    """Get assertion information.
+
+    :param assertion_params: a sequence of strings to pass to 'snap known'.
+    :returns: a stream of bytes from the assertion.
+    :rtype: bytes
+    """
+    try:
+        return check_output(["snap", "known", *assertion_params])
+    except CalledProcessError as call_error:
+        raise errors.SnapGetAssertionError(
+            assertion_params=assertion_params
+        ) from call_error
+
+
+def _get_parsed_snap(snap: str) -> Tuple[str, str]:
+    if "/" in snap:
```

Comment:
we had a long standing request to change the snap name and channel separator to be an =, mind checking with @mvo5 if that is still the case?